pkg/kms/webkms/remotekms.go

```diff
@@ -0,0 +1,297 @@
+	if err != nil {
+		return "", fmt.Errorf("failed to save keystoreURL in remoteKMS configStore: %w", err)
+	}
+
+	// the following can be stored earlier, but storing them after keystoreURL to ensure it was created
+	err = store.Put(ControllerField, []byte(controller))
+	if err != nil {
+		return "", fmt.Errorf("failed to save Controller in remoteKMS configStore: %w", err)
+	}
+
+	err = store.Put(SecretField, []byte(base64.URLEncoding.EncodeToString(secret)))
+	if err != nil {
+		return "", fmt.Errorf("failed to save secret in remoteKMS configStore: %w", err)
+	}
+
+	return keystoreURL, nil
+}
+
+// Create a new key/keyset/key handle for the type kt remotely
+// Returns:
+//  - KeyID raw ID of the handle
+//  - handle instance representing a remote keystore URL including KeyID
+//  - error if failure
+func (r *RemoteKMS) Create(kt kms.KeyType) (string, interface{}, error) {
+	destination := r.keystoreURL + "/keys"
+	httpReq := &createKeyReq{
+		KeyType:    string(kt),
+		Passphrase: r.secret,
+	}
+
+	req, err := json.Marshal(httpReq)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to marshal Create key request [%s, %w]", destination, err)
+	}
+
+	resp, err := r.httpClient.Post(destination, ContentType, bytes.NewBuffer(req))
+	if err != nil {
+		return "", nil, fmt.Errorf("posting Create key failed [%s, %w]", destination, err)
+	}
+
+	// handle response
+	defer closeResponseBody(resp.Body, logger, "Create")
+
+	kidURL := resp.Header.Get(LocationHeader)
+	kid := kidURL[strings.LastIndex(kidURL, "/")+1:]
+
+	return kid, kidURL, nil
+}
+
+// Get key handle for the given KeyID remotely
+// Returns:
+//  - handle instance representing a remote keystore URL including KeyID
+//  - error if failure
+func (r *RemoteKMS) Get(keyID string) (interface{}, error) {
+	return r.buildKIDURL(keyID)
+}
+
+func (r *RemoteKMS) buildKIDURL(keyID string) (string, error) {
+	keystoreURL, err := r.configStore.Get(KeystoreURLField)
+	if err != nil {
+		return "", err
+	}
+
+	return string(keystoreURL) + "/keys/" + keyID, nil
+}
+
+// Rotate remotely a key referenced by KeyID and return a new handle of a keyset including old key and
+// new key with type kt. It also returns the updated KeyID as the first return value
+// Returns:
+//  - new KeyID
+//  - handle instance (to private key)
+//  - error if failure
+func (r *RemoteKMS) Rotate(kt kms.KeyType, keyID string) (string, interface{}, error) {
+	return "", nil, errors.New("function Rotate is not implemented in remoteKMS")
+}
+
+// ExportPubKeyBytes will remotely fetch a key referenced by id then gets its public key in raw bytes and returns it.
+// The key must be an asymmetric key.
+// Returns:
+//  - marshalled public key []byte
+//  - error if it fails to export the public key bytes
+func (r *RemoteKMS) ExportPubKeyBytes(keyID string) ([]byte, error) {
+	keyURL, err := r.buildKIDURL(keyID)
+	if err != nil {
+		return nil, err
+	}
+
+	destination := keyURL + "/keys/export"
```

Comment:
endpoint is going to be "/keys/" + keyID + "/export"

Reply:
will update..

Reply:
done